### PR TITLE
VMWare NSX Plugin

### PR DESCRIPTION
**Pull request by nathan-rightscale**
### Description
Initial Release of VMWare NSX Plugin

### Contribution Check List

- [X] All tests pass.
- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: vmware/nsx/README.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+1. Create RightScale Credentials with values that match the NSX User (Credential name: `NSX_USER`) & NSX Password (Credential name: `NSX_PASSWORD`) that will be used by Self-Service to interact with NSX. This user must have permissions required by the CloudApplication. 
+1. Update the default_host and path of the nsx_plugin.cat.rb to include the host where j2xrp resides and the _token value generated in the wstunnel setup.
+1. Navigate to the appropriate Self-Service portal
+   - For more details on using the portal review the [SS User Interface Guide](http://docs.rightscale.com/ss/guides/ss_user_interface_guide.html)
+1. In the Design section, use the `Upload CAT` interface to complete the following:
+   1. Upload each of packages listed in the Requirements Section
+   1. Upload the `nsx_plugin.cat.rb` file located in this repository
+ 
+## How to Use
+The Azure Compute Plugin has been packaged as `plugins/rs_vmware_nsx`. In order to use this plugin you must import this plugin into a CAT.
```
Comment (rshade) [inline vmware/nsx/README.md]: change from Azure Compute to NSX

File: vmware/nsx/nsx_plugin.cat.rb [hunk 2]
```diff
@@ -0,0 +1,813 @@
+name 'VMware NSX Plugin'
+short_description 'A RightScale Self-Service plugin for VMware NSX.'
+rs_ca_ver 20161221
```
Comment (rshade) [inline vmware/nsx/nsx_plugin.cat.rb]: can you add a version in the Long Description

File: vmware/nsx/nsx_plugin.cat.rb [hunk 3]
```diff
@@ -0,0 +1,813 @@
...
+
+    field 'dynamicMemberDefinition' do
+      type 'object'
+    end
+
+    output_path '/securitygroup'
+    
+    output 'objectId','objectTypeName','revision','description','name'
+
+    # output 'objectId'
```
Comment (rshade) [inline vmware/nsx/nsx_plugin.cat.rb]: can you clean these out?

File: vmware/nsx/server_templates/NSX - Firewall Demo/RL10_Linux_Enable_Managed_Login.sh [hunk 4]
```diff
@@ -0,0 +1,323 @@
+#!/bin/bash
```
Comment (rshade) [inline vmware/nsx/server_templates/NSX - Firewall Demo/RL10_Linux_Enable_Managed_Login.sh]: can the rightlink scripts point to the real ones?

File: vmware/nsx/README.md (hunk 1 above)
Comment (nathan-rightscale) [reply vmware/nsx/README.md]: Done

File: vmware/nsx/nsx_plugin.cat.rb (hunk 2 above)
Comment (nathan-rightscale) [reply vmware/nsx/nsx_plugin.cat.rb]: Done

File: vmware/nsx/nsx_plugin.cat.rb (hunk 3 above)
Comment (nathan-rightscale) [reply vmware/nsx/nsx_plugin.cat.rb]: Done

File: vmware/nsx/server_templates/NSX - Firewall Demo/RL10_Linux_Enable_Managed_Login.sh (hunk 4 above)
Comment (nathan-rightscale) [reply vmware/nsx/server_templates/NSX - Firewall Demo/RL10_Linux_Enable_Managed_Login.sh]: Done

**Pull request merged**